### PR TITLE
add ignore no primary key table param

### LINKED ISSUE #3006: [enhancement] [ cdc-connector] [ignore no primary key table] When using regular expressions to synchronize the entire database with Flink CDC, skip tables that do not have a primary key.
### Search before asking

- [x] I searched in the [issues](https://github.com/ververica/flink-cdc-connectors/issues) and found nothing similar.

### Motivation

If regular expressions are used to match the tables to be synchronized, and some tables in the database do not have a primary key, when submitting a task with Flink CDC 3.0, an error occurs. Can we add an option to skip the synchronization of tables without a primary key, such as ignore-no-primary-key-table: true?

**Flink version**
1.18

**Flink CDC version**
3.0.0

**Database and its version**
Mysql 5.7 ,  StarRocks 3.2.2

### Solution

ignore-no-primary-key-table: true

### Alternatives

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by dufeng1010**
add ignore.noprimarykey.table param, default this param is false, but when this param is set to true, it can ignore no primary key tables.
fix #3006

### CODE REVIEW

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/com/ververica/cdc/connectors/mysql/utils/MySqlSchemaUtils.java [hunk 1]
```diff
@@ -117,6 +118,56 @@ public static List<TableId> listTables(JdbcConnection jdbc, String dbName) throw
...
+     */
+    public static List<TableId> listPrimaryKeyTables(JdbcConnection jdbc, String dbName)
+            throws SQLException {
+
+        final List<TableId> tableIds = new ArrayList<>();
+
+        jdbc.query(
+                "select distinct TABLE_NAME from INFORMATION_SCHEMA.COLUMNS where TABLE_SCHEMA = '"
+                        + dbName
+                        + "' and COLUMN_KEY = 'PRI'",
```
Comment (lvyanquan) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/com/ververica/cdc/connectors/mysql/utils/MySqlSchemaUtils.java:159]: will the following sql be better?
```
SELECT DISTINCT TABLE_NAME 
FROM
    INFORMATION_SCHEMA.TABLE_CONSTRAINTS
WHERE
    CONSTRAINT_TYPE = 'PRIMARY KEY'
    AND TABLE_SCHEMA = 'your_database_name';
```

**Comment by lvyanquan** [comment]
Can you add a test like [testCreateSource](https://github.com/ververica/flink-cdc-connectors/blob/7ea197242f82d1f63f0e10c6c79ed283f2f7adb7/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/com/ververica/cdc/connectors/mysql/source/MySqlDataSourceFactoryTest.java#L46)?

**Review by everhopingandwaiting — APPROVED** [review]
ok

**Comment by gong** [comment]
@dufeng1010 I think we should log print ignored table name. Beacause it can help user to  check config. And add doc for new add param

**Comment by ruanhang1993** [comment]
> @dufeng1010 I think we should log print ignored table name. Beacause it can help user to check config. And add doc for param `ignore.noprimarykey.table`

@dufeng1010 +1 for this. I think it is better to print the filtered tables in the log.

**Comment by everhopingandwaiting** [comment]
> > @dufeng1010 I think we should log print ignored table name. Beacause it can help user to check config. And add doc for param `ignore.noprimarykey.table`
> 
> @dufeng1010 +1 for this. I think it is better to print the filtered tables in the log.

282d9ad583304f458fc34139b5582a30

there are already print all available tables, but it`s better to print the filtered tables in the log

**Comment by dufeng1010** [comment]
i will fix it.

**Review by everhopingandwaiting — APPROVED** [review]
LGTM

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/com/ververica/cdc/connectors/mysql/utils/MySqlSchemaUtils.java [hunk 2]
```diff
@@ -117,6 +118,114 @@ public static List<TableId> listTables(JdbcConnection jdbc, String dbName) throw
...
+                "select tab.table_name\n"
+                        + " from information_schema.tables tab\n"
+                        + " left join information_schema.table_constraints tco\n"
+                        + "          on tab.table_schema = tco.table_schema\n"
+                        + "          and tab.table_name = tco.table_name\n"
+                        + "          and tco.constraint_type = 'PRIMARY KEY'\n"
+                        + " where tco.constraint_type is null\n"
+                        + "      and tab.table_type = 'BASE TABLE'\n"
+                        + "      and tab.table_schema = '"
+                        + dbName
```
Comment (lvyanquan) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/com/ververica/cdc/connectors/mysql/utils/MySqlSchemaUtils.java:216]: Hi @dufeng1010 , I wonder why this SQL is quiet different with the listPrimaryKeyTables method.

Comment (dufeng1010) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/com/ververica/cdc/connectors/mysql/utils/MySqlSchemaUtils.java:216]: this method is get all tables without primary key. 
listPrimarykeyTables method is get all tables with primary key

Comment (lvyanquan) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/com/ververica/cdc/connectors/mysql/utils/MySqlSchemaUtils.java:216]: Get it. It's better to unify the case.

**Pull request closed without merging**